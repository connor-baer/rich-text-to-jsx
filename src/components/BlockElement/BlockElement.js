import React, { Fragment } from 'react';

/**
 * Default fallback element for block level nodes.
 * Renders a prominent warning in development.
 */
function BlockElement(props) {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line no-console
    console.debug('Unknown block  node type', props);
    return (
      <Fragment>
        <span style={{ color: 'white', backgroundColor: 'red' }}>
          Unknown block node type.
        </span>
        {props.children}
      </Fragment>
    );
  }
  return props.children || null;
}

if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  const PropTypes = require('prop-types');

  BlockElement.propTypes = {
    /**
     * The element children.
     */
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.array]),
  };
}

/**
 * @component
 */
export default BlockElement;
