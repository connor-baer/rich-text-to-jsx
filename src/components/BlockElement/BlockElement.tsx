import React, { Fragment } from 'react';

export interface BlockElementProps {
  /**
   * The element children.
   */
  children?: JSX.Element;
}

/**
 * Default fallback element for block level nodes.
 * Renders a prominent warning in development.
 */
export default function BlockElement(
  props: BlockElementProps,
): JSX.Element | null {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line no-console
    console.debug('Unknown block  node type', props);
    return (
      <Fragment>
        <span style={{ color: 'white', backgroundColor: 'red' }}>
          Unknown block node type.
        </span>
        {props.children}
      </Fragment>
    );
  }
  return props.children || null;
}
