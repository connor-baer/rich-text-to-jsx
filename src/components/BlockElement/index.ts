import BlockElement from './BlockElement';

export default BlockElement;
