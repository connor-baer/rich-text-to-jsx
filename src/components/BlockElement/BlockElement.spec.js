import React from 'react';
import { create } from 'react-test-renderer';

import BlockElement from './BlockElement';

/* eslint-disable no-console */

describe('BlockElement', () => {
  describe('in development', () => {
    const nodeEnv = process.env.NODE_ENV;
    const { debug } = console;

    beforeAll(() => {
      console.debug = jest.fn();
      process.env.NODE_ENV = 'development';
    });

    afterAll(() => {
      console.debug = debug;
      process.env.NODE_ENV = nodeEnv;
    });

    it('should log a debug message to the console', () => {
      create(<BlockElement />);
      expect(console.debug).toHaveBeenCalledTimes(1);
    });

    it('should render a warning', () => {
      const actual = create(<BlockElement />);
      expect(actual).toMatchSnapshot();
    });

    it('should render a warning and the children', () => {
      const actual = create(
        <BlockElement>
          <p>Foo</p>
        </BlockElement>,
      );
      expect(actual).toMatchSnapshot();
    });
  });

  describe('in production', () => {
    const nodeEnv = process.env.NODE_ENV;

    beforeAll(() => {
      process.env.NODE_ENV = 'production';
    });

    afterAll(() => {
      process.env.NODE_ENV = nodeEnv;
    });

    it('should render null', () => {
      const actual = create(<BlockElement />);
      expect(actual).toMatchSnapshot();
    });

    it('should render the children', () => {
      const actual = create(
        <BlockElement>
          <p>Bar</p>
        </BlockElement>,
      );
      expect(actual).toMatchSnapshot();
    });
  });
});
