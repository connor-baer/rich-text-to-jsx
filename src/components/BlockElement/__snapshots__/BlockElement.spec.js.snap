// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockElement in development should render a warning 1`] = `
<span
  style="color: white; background-color: red;"
>
  Unknown block node type.
</span>
`;

exports[`BlockElement in development should render a warning and the children 1`] = `
<div>
  <span
    style="color: white; background-color: red;"
  >
    Unknown block node type.
  </span>
  <p>
    Foo
  </p>
</div>
`;

exports[`BlockElement in production should render null 1`] = `null`;

exports[`BlockElement in production should render the children 1`] = `
<p>
  Bar
</p>
`;
